azure/services/virtualmachines/client.go

```diff
@@ -38,225 +35,144 @@ import (
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create armcompute client factory")
+	}
+	return &AzureClient{factory.NewVirtualMachinesClient()}, nil
 }
 
 // Get retrieves information about the model view or the instance view of a virtual machine.
 func (ac *AzureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (result interface{}, err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.Get")
 	defer done()
 
-	return ac.virtualmachines.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), "")
+	resp, err := ac.virtualmachines.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), nil)
+	if err != nil {
+		return nil, err
+	}
+	return resp.VirtualMachine, nil
 }
 
 // GetByID retrieves information about the model or instance view of a virtual machine.
-func (ac *AzureClient) GetByID(ctx context.Context, resourceID string) (compute.VirtualMachine, error) {
+func (ac *AzureClient) GetByID(ctx context.Context, resourceID string) (armcompute.VirtualMachine, error) {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.GetByID")
 	defer done()
 
 	parsed, err := azureutil.ParseResourceID(resourceID)
 	if err != nil {
-		return compute.VirtualMachine{}, errors.Wrap(err, fmt.Sprintf("failed parsing the VM resource id %q", resourceID))
+		return armcompute.VirtualMachine{}, errors.Wrap(err, fmt.Sprintf("failed parsing the VM resource id %q", resourceID))
 	}
 
 	log.V(4).Info("parsed VM resourceID", "parsed", parsed)
 
-	return ac.virtualmachines.Get(ctx, parsed.ResourceGroupName, parsed.Name, "")
+	result, err := ac.Get(ctx, newResourceAdaptor(parsed))
+	if err != nil {
+		return armcompute.VirtualMachine{}, err
+	}
+
+	if vm, ok := result.(armcompute.VirtualMachine); ok {
+		return vm, nil
+	}
+	return armcompute.VirtualMachine{}, errors.Errorf("expected VirtualMachine but got %T", result)
 }
 
 // CreateOrUpdateAsync creates or updates a virtual machine asynchronously.
-// It sends a PUT request to Azure and if accepted without error, the func will return a Future which can be used to track the ongoing
+// It sends a PUT request to Azure and if accepted without error, the func will return a Poller which can be used to track the ongoing
 // progress of the operation.
-func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, parameters interface{}) (result interface{}, future azureautorest.FutureAPI, err error) {
+func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, resumeToken string, parameters interface{}) (result interface{}, poller *runtime.Poller[armcompute.VirtualMachinesClientCreateOrUpdateResponse], err error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.CreateOrUpdate")
 	defer done()
 
-	vm, ok := parameters.(compute.VirtualMachine)
+	vm, ok := parameters.(armcompute.VirtualMachine)
 	if !ok {
```

Comment:
Actually I think the CI errors are failing without this change we've been making to all the services: 
```suggestion
	if !ok && parameters != nil {
```